src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MovieRecommender {
+    String path;
+    int repMovies,repUsers;
+    int user,product,review;
+    HashMap<String, Integer> userId = new HashMap<String, Integer>();
+    HashMap<String, Integer> products = new HashMap<String, Integer>();
+    HashMap<Integer,String> productId = new HashMap<Integer,String>();
+
+    public MovieRecommender(String pathUrl) throws IOException {
+        this.path = pathUrl;
+        convert();
+    }
+
+    public void convert() throws IOException {
+        this.user=0;
+        this.product=0;
+        this.review=0;
+        this.repMovies=0;
+        this.repUsers=0;
+        String idUser="",idProduct="", scoreReview="";
+
+        BufferedReader reader = new BufferedReader(new FileReader(path));
+        String line;
+
+        String uRegex = "review\\/userId: ([\\D\\d]+)";
+        String pRegex = "product\\/productId: ([\\D\\d]+)";
+        String rRegex = "review\\/score: ([\\D\\d]+)";
+
+        Pattern userPattern = Pattern.compile(uRegex);
+        Pattern productPattern = Pattern.compile(pRegex);
+        Pattern reviewPattern = Pattern.compile(rRegex);
+
+
+        FileWriter csvWriter = new FileWriter("list.csv");
+
+        Matcher matcher;
+
+        boolean match;
+
+        line= reader.readLine();
+
+        while (line != null){
+
+            matcher = productPattern.matcher(line);
+            match = matcher.matches();
+
+            if (match){
+                String[] parts = line.split(": ",2);
+                idProduct= parts[1];
+                if(products.containsKey(idProduct))
+                    this.repMovies++;
+                else{
+                    this.product++;
+                    products.put(idProduct,this.product);
+                    productId.put(this.product,idProduct);
+                }
+
+            }
+
+            matcher = userPattern.matcher(line);
+            match = matcher.matches();
+
+            if (match){
+                String[] parts = line.split(": ",2);
+                idUser= parts[1];
+                if(userId.containsKey(idUser))
+                    this.repUsers++;
+                else{
+                    this.user++;
+                    userId.put(idUser,this.user);
+                }
+
+            }
+
+            matcher = reviewPattern.matcher(line);
+            match = matcher.matches();
+
+            if (match){
```

Comment:
I see three different if statements for matches, does it mean that a single line can match with all three patterns at the same time?